src/fcntl.rs

```diff
@@ -343,7 +343,11 @@ fn inner_readlink<P: ?Sized + NixPath>(
     dirfd: Option<RawFd>,
     path: &P,
 ) -> Result<OsString> {
-    let mut v = Vec::with_capacity(libc::PATH_MAX as usize);
+    #[cfg(not(target_os = "hurd"))]
+    const PATH_MAX: usize = libc::PATH_MAX as usize;
+    #[cfg(target_os = "hurd")]
+    const PATH_MAX: usize = 1024;
```

Comment:
Is `PATH_MAX` really not defined in Hurd, or does libc just need a new PR for it?

Reply:
It is really not defined on GNU/Hurd

Reply:
Would you please leave a comment explaining why we are using a magic number here? Something like "Hurd does not have `PATH_MAX` defined, so we have to use a magic number here"

Reply:
(done so)